src/aeso_fcode_to_fate.erl

```diff
@@ -240,14 +240,17 @@ term_to_fate({lit, L}) ->
         {contract_pubkey, K} -> aeb_fate_data:make_contract(K);
         {oracle_pubkey, K}   -> aeb_fate_data:make_oracle(K);
         {oracle_query_id, _} -> ?TODO(fate_oracle_query_id_value)
-     end;
+     end.
+
+term_to_fate({lit, L}) ->
+    lit_to_fate(L);
 %% negative literals are parsed as 0 - N
 term_to_fate({op, '-', [{lit, {int, 0}}, {lit, {int, N}}]}) ->
     aeb_fate_data:make_integer(-N);
 term_to_fate(nil) ->
     aeb_fate_data:make_list([]);
-term_to_fate({op, '::', [Hd | Tl]}) ->
-    aeb_fate_data:make_list([term_to_fate(Hd) | [ term_to_fate(E) || E<-Tl]]);
+term_to_fate({op, '::', [Hd, Tl]}) ->
+    aeb_fate_data:make_list([term_to_fate(Hd) | aeb_fate_data:make_list(term_to_fate(Tl))]);
```

Comment:
The second `make_list` makes no sense. `term_to_fate` already adds a `make_list` so if anything that `make_list` should be stripped off (but `make_list` is currently a no-op, so...). I guess the most correct thing to do without messing with aebytecode (and adding `aeb_fate_data:X_value` functions) is to include `aeb_fate_data.hrl` and use ?FATE_LIST_VALUE.

Reply:
In that case I probably would just only use `term_to_fate(Tl)` because including that abstraction breaker in this code seems weird.

Reply:
Remove the outer `make_list` as well then.